src/transformers/models/auto/auto_factory.py

```diff
@@ -377,13 +390,26 @@ def from_config(cls, config, **kwargs):
     @classmethod
     def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         config = kwargs.pop("config", None)
+        trust_remote_code = kwargs.pop("trust_remote_code", False)
         kwargs["_from_auto"] = True
         if not isinstance(config, PretrainedConfig):
             config, kwargs = AutoConfig.from_pretrained(
                 pretrained_model_name_or_path, return_unused_kwargs=True, **kwargs
             )
-
-        if type(config) in cls._model_mapping.keys():
+        if hasattr(config, "auto_map") and cls.__name__ in config.auto_map:
+            if not trust_remote_code:
```

Comment:
I wonder if it wouldn't be good practice to force a revision here too. Either through raising an error (best) or through logging a warning (ok): if a public repository becomes trendy, it wouldn't be surprising to see the user update the code and for it to, at best break systems running the code, at worst run arbitrary code through `torch.load`

Reply:
That would make the API harder to use, no?

Reply:
After discussing with @LysandreJik, we will need to remember to parse this `auto_map` attribute in the hub so we know which models are custom code models